### PR TITLE
IGNITE-7821 Update Apache Ignite and Web Console Dockerfile for 2.4 version and local build

### PR DESCRIPTION
**Pull request by vveider**
* removed Apache Ignite's Docker module obsolete autobuild directory layout
 * updated Apache Ignite's Dockerfile for local image build and use of lighter base image
 * removed Web Console's build script - replaced with pre-build instructions
 * introduced minor refactoring to Web Console's Dockerfile
 * fixed .dockerignore file with correct paths

### CODE REVIEW

File: modules/web-console/docker/standalone/Dockerfile [hunk 1]
```diff
@@ -63,22 +45,22 @@ WORKDIR /opt/web-console
...
+    npm install --only=production --no-optional
 
 # Copy nginx config.
-COPY ./nginx/nginx.conf /etc/nginx/nginx.conf
-COPY ./nginx/web-console.conf /etc/nginx/web-console.conf
+COPY nginx/* /etc/nginx/
 
-# Setup entrypoint.
-COPY ./entrypoint.sh .
-RUN chmod 755 /opt/web-console/entrypoint.sh
```
Comment (nva) [inline modules/web-console/docker/standalone/Dockerfile]: Is this correct to remove this line if permissions on host machine is incorrect?

File: modules/web-console/docker/standalone/Dockerfile [hunk 2]
```diff
@@ -63,22 +45,22 @@ WORKDIR /opt/web-console
...
-RUN cd /opt/web-console/frontend && npm install --no-optional --prod && npm run build
-RUN cd /opt/web-console/backend && npm install --only=production --no-optional
+RUN cd /opt/web-console/frontend && \
+    npm install --no-optional --prod && \
+    npm run build
 
-# Returns to base path.
-WORKDIR /opt/web-console
+RUN cd /opt/web-console/backend && \
+    npm install --only=production --no-optional
```
Comment (nva) [inline modules/web-console/docker/standalone/Dockerfile]: Please, use the same command format as before (npm install --no-optional --prod).

File: modules/web-console/docker/standalone/Dockerfile [hunk 3]
```diff
@@ -18,39 +18,21 @@
...
-    done
+ENV NODE_VERSION 8.10.0
 
 RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv EA312927 && \
-   echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.2 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.2.list
+    echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.2 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.2.list
 
 # Update package list & install.
 RUN apt-get update && \
     apt-get install -y nginx-light mongodb-org-server curl xz-utils git
```
Comment (nva) [inline modules/web-console/docker/standalone/Dockerfile]: Is you really need "git" package?

File: modules/web-console/docker/standalone/Dockerfile (hunk 1 above)
Comment (vveider) [reply modules/web-console/docker/standalone/Dockerfile]: As I see under Windows OS, executable flag exists, so there should be no problems.
```MINGW64 /c/Personal/code/incubator-ignite/modules/web-console/docker/standalone (ignite-7821)
$ git ls-files -s entrypoint.sh
100755 3882f06a2ed5e91006e3f94a3b1cca749b0ec10a 0       entrypoint.sh```

**Comment by dspavlov** [comment]
The ticket is resolved. Closing PR

**Pull request closed without merging**